src/bin/genemichaels.rs

```diff
@@ -196,13 +232,103 @@ fn main() {
             Offable::On(_) => true,
         },
     };
+    if args.workspace {
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
 
-    fn skip(src: &str) -> bool {
-        src.lines().take(5).any(|l| l.contains("`nogenemichaels`"))
-    }
+                // bold green
+                eprint!("\x1B[1;32m");
+                eprint!("  Formatting ");
 
-    if args.files.is_empty() {
-        match es!({
+                // reset
+                eprint!("\x1B[0;22m");
+                eprintln!("workspace...");
+                let mut cargo_toml = None;
+
+                // find workspace cargo toml
+                for parent in 0.. {
```

Comment:
I think this never terminates if it can't find Cargo.toml.  Could this be done with the current path and `.parent()` somehow?

Reply:
It `panic!()`'s eventually - we could use panic unwind here to catch it if desired happy to take on suggestions here.

Reply:
Maybe something like this?

```
let cdir = current_dir()?;
let mut at: Option<&Path> = Some(&cdir);
while let Some(d) = at.take() {
    if d.join("Cargo.toml").exists() {
        cargo_toml = Some(d);
        break;
    }
    at = d.parent();
}
```